frontend/src/components/App/Settings/SettingsCluster.tsx

```diff
@@ -222,200 +232,205 @@ export default function SettingsCluster() {
-                    let value = event.target.value;
-                    value = value.replace(' ', '');
-                    setUserDefaultNamespace(value);
-                  }}
-                  value={userDefaultNamespace}
-                  placeholder={defaultNamespace}
-                  error={!isValidDefaultNamespace}
-                  helperText={
-                    isValidDefaultNamespace
-                      ? t(
-                          'translation|The default namespace for e.g. when applying resources (when not specified directly).'
-                        )
-                      : invalidNamespaceMessage
-                  }
-                  InputProps={{
-                    endAdornment: isEditingDefaultNamespace() ? (
-                      <Icon
-                        width={24}
-                        color={theme.palette.text.secondary}
-                        icon="mdi:progress-check"
-                      />
-                    ) : (
-                      <Icon width={24} icon="mdi:check-bold" />
-                    ),
-                    sx: { maxWidth: 250 },
-                  }}
-                />
-              ),
-            },
-            {
-              name: t('translation|Allowed namespaces'),
-              value: (
-                <>
-                  <TextField
-                    onChange={event => {
-                      let value = event.target.value;
-                      value = value.replace(' ', '');
-                      setNewAllowedNamespace(value);
-                    }}
-                    placeholder="namespace"
-                    error={!isValidNewAllowedNamespace}
-                    value={newAllowedNamespace}
-                    helperText={
-                      isValidNewAllowedNamespace
-                        ? t(
-                            'translation|The list of namespaces you are allowed to access in this cluster.'
-                          )
-                        : invalidNamespaceMessage
-                    }
-                    autoComplete="off"
-                    inputProps={{
-                      form: {
-                        autocomplete: 'off',
-                      },
-                    }}
-                    InputProps={{
-                      endAdornment: (
-                        <IconButton
-                          onClick={() => {
-                            storeNewAllowedNamespace(newAllowedNamespace);
-                          }}
-                          disabled={!newAllowedNamespace}
-                          size="medium"
-                        >
-                          <InlineIcon icon="mdi:plus-circle" />
-                        </IconButton>
-                      ),
-                      onKeyPress: event => {
-                        if (event.key === 'Enter') {
-                          storeNewAllowedNamespace(newAllowedNamespace);
+        {cluster ? (
```

Comment:
Instead of adding this big conditional to the block, you can use what was there before (checking for cluster before) and return the empty component instead of the null that was being returned before. This will allow the changes to be less so.

Reply:
Reverted the change